kubernetes/userspace-daemonset.yml

```diff
@@ -0,0 +1,29 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: userspace-cni
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: userspace-cni
+  template:
+    metadata:
+      labels:
+        app: userspace-cni
+    spec:
+      initContainers:
+      - name: userspace-cni-plugin
+        image: localhost:5000/userspacecni #registory:imagename
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: cnibin
+          mountPath: /opt/cni/bin
+      volumes:
+        - name: cnibin
+          hostPath:
+            path: /opt/cni/bin
+      containers:
```

Comment:
so the init container copies the binary, and then it goes into the main container, which is a pause container, thus releasing resources.
correct?